src/main/java/org/openstreetmap/atlas/checks/flag/FlaggedRelation.java

```diff
@@ -0,0 +1,146 @@
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasItem;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.LocationItem;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMemberList;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A flag for a {@link Relation}
+ *
+ * @author sayas01
+ */
+public class FlaggedRelation extends FlaggedObject
+{
+    private static final Logger logger = LoggerFactory.getLogger(FlaggedPoint.class);
+    private final Relation relation;
+    private final Map<String, String> properties;
+
+    public FlaggedRelation(final Relation relation)
+    {
+        this.relation = relation;
+        this.properties = initProperties(relation);
+    }
+
+    /**
+     * Get locations of all flattened members
+     *
+     * @return flagged geometry
+     */
+    @Override
+    public Iterable<Location> getGeometry()
+    {
+        final List<Location> listOfLocation = new ArrayList<>();
+
+        final Set<AtlasObject> flattenedMembers = this.relation.flatten();
+        if (flattenedMembers.size() == 1)
+        {
+            final AtlasObject object = flattenedMembers.iterator().next();
+            if (object instanceof LocationItem)
+            {
+                return ((LocationItem) object).getLocation();
+            }
+        }
+        for (final AtlasObject member : flattenedMembers)
+        {
+            if (member instanceof LocationItem)
+            {
+                listOfLocation.add(((LocationItem) member).getLocation());
+            }
+            else
+            {
+                if (member instanceof Area)
+                {
+                    // An Area's geometry doesn't include the end (same as start) location. To close
+                    // the
+                    // area boundary, we need to add the end location manually.
+                    final List<Location> geometry = Iterables
+                            .asList(((AtlasItem) member).getRawGeometry());
+                    geometry.add(geometry.get(0));
+                    listOfLocation.addAll(new PolyLine(geometry));
+                }
+                else
+                {
+                    listOfLocation.addAll(new PolyLine(((AtlasItem) member).getRawGeometry()));
+                }
+            }
+        }
+        final PolyLine polyLine = new PolyLine(listOfLocation);
+        return polyLine;
+    }
+
+    /**
+     * Get flattened RelationMemberList
+     * 
+     * @return {@link RelationMemberList}
+     */
+    public RelationMemberList getFlattenedRelationMembers()
+    {
+        final List<RelationMember> listOfMembers = new ArrayList<>();
```

Comment:
Well done on the graph traversal! Is there a reason you've implemented this by hand instead of using the relation.flatten() function?

Reply:
Yes, `relation.flatten()` flattens it to a list of AtlasObject  where as `getFlattenedRelationMembers()` returns a RelationMemberList which has information of the relation, the member belongs to and also its role. This information is later used in the code to update properties of the flagged members.

Reply:
Gotcha -- and getting that information from an `AtlasObject` would be pretty gnarly, so I'm fine with leaving this as is :)